eslint/babel-eslint-plugin/test/rules/no-invalid-this.js

```diff
@@ -78,40 +89,94 @@ function extractPatterns(patterns, type) {
 //------------------------------------------------------------------------------
 
 const patterns = [
-  // Class private fields
   {
-    code: "class A {a = this.b;};",
-    parserOptions: { ecmaVersion: 6 },
-    valid: [NORMAL, USE_STRICT, IMPLIED_STRICT, MODULES],
+    code: "class A { a = this.b; }",
+    parserOptions: { ecmaVersion: 2020 },
+    valid: [MODULE, SCRIPT, USE_STRICT, IMPLIED_STRICT],
     invalid: [],
   },
-
   {
-    code: "class A {a = () => {return this.b;};};",
-    parserOptions: { ecmaVersion: 6 },
-    valid: [NORMAL, USE_STRICT, IMPLIED_STRICT, MODULES],
+    code: "class A { a = () => { return this.b; }; }",
+    parserOptions: { ecmaVersion: 2020 },
+    valid: [MODULE, SCRIPT, USE_STRICT, IMPLIED_STRICT],
     invalid: [],
   },
-
-  // Class Private methods
   {
-    code: "class A {#a = this.b;};",
-    parserOptions: { ecmaVersion: 6 },
-    valid: [NORMAL, USE_STRICT, IMPLIED_STRICT, MODULES],
+    code: "class A { a = () => this.b; }",
+    parserOptions: { ecmaVersion: 2020 },
+    valid: [MODULE, SCRIPT, USE_STRICT, IMPLIED_STRICT],
     invalid: [],
   },
+  {
+    code: "class A { [this] = b; }",
+    parserOptions: { ecmaVersion: 2020 },
+    valid: [],
+    invalid: [MODULE, SCRIPT, USE_STRICT, IMPLIED_STRICT],
+  },
+  {
+    code: "class A { [this.a] = b; }",
+    parserOptions: { ecmaVersion: 2020 },
+    valid: [],
+    invalid: [MODULE, SCRIPT, USE_STRICT, IMPLIED_STRICT],
+  },
+  {
+    code: "class A { [`${this.a}b`] = c; }",
+    parserOptions: { ecmaVersion: 2020 },
+    valid: [],
+    invalid: [MODULE, SCRIPT, USE_STRICT, IMPLIED_STRICT],
+  },
+  {
+    code: "class A { [this.a] = () => { return b; }; }",
+    parserOptions: { ecmaVersion: 2020 },
+    valid: [],
+    invalid: [MODULE, SCRIPT, USE_STRICT, IMPLIED_STRICT],
+  },
+  {
+    code: "class A { [this.a] = () => b; }",
+    parserOptions: { ecmaVersion: 2020 },
+    valid: [],
+    invalid: [MODULE, SCRIPT, USE_STRICT, IMPLIED_STRICT],
+  },
+  {
+    code: "class A { bFactory = () => class B { [this.b] = () => c; }; }",
```

Comment:
This is a valid case, the `this.b` will be bound to the instance of `A` since it is in the initializer of class property.

Reply:
That makes sense! Thanks for catching this.